doc/synapse-unit-testing-framework.md

```diff
@@ -0,0 +1,375 @@
+# Synapse Unit Testing Framework
+Unit testing framework for synapse provides an in-built way to execute a unit test against a given synapse configuration artifacts such as sequence, proxy, api, local entry, endpoint. Mainly it required a test artifact which is your main configuration, supportive artifacts which needs to execute the test artifact if exists and test cases with assertions. 
+
+### Features
+- Support deploy sequence, proxy, API, endpoint and local-entry artifacts
+- Able to run multiple test cases against a test artifact and supportive artifacts
+- Able to create, start and stop mock-services with multiple resources
+- Assertion with `AssertEquals` and `AssertNotNull` properties
+- Assertion for response body(`$body`), message context property(`$ctx`), axis2 property(`$axis2`) and transport property(`$trp`) values
+
+### Running Unit Testing Framework
+
+You can use the following command depending on the platform to start the unit testing framework in the Micro Integrator.
+- MacOS/Linux/CentOS - `sh <MI_HOME>/bin/micro-integrator.sh -synapseTest <port>`
+- Windows - `<MI_HOME>/bin/micro-integrator.bat -synapseTest <port>`
+
+By default the unit testing framework starts on port 9007 and you can change the starting port by passing the port as above
```

Comment:
Please note that there are few samples that will conflict with this port number. We either have to use a different port for this or use different ports in samples.

1. https://docs.wso2.com/display/EI640/Sample+58%3A+Static+Load+Balancing+between+Three+Nodes

Reply:
It would be better to use different port than modifying the samples

Reply:
What if we define the default port in the startup script itself when there is no second argument instead of defining in the synapse. @a5anka any thoughts ?

Reply:
Yes, it is possible. Then we need to remove the "set unit test default port" part from the synapse source code.

Reply:
:+1:  for the idea

Reply:
Done